src/BigBlueButton.php

```diff
@@ -90,6 +97,51 @@ public function getApiVersion()
         return new ApiVersionResponse($xml);
     }
 
+    /**
+     * Check if connection to api can be established with the baseurl and secret
+     * @return bool connection successful
+     */
+    public function isConnectionWorking(): bool
+    {
+        $this->connectionError = null;
+        try {
+            $response = $this->isMeetingRunning(
+                new IsMeetingRunningParameters("connection_check")
+            );
+
+            // invalid url
+            if (!$response->success() && !$response->failed()) {
+                $this->connectionError = self::CONNECTION_ERROR_BASEURL;
+                return false;
+            }
+
+            // invalid secret
+            if (!$response->success()) {
+                $this->connectionError = self::CONNECTION_ERROR_SECRET;
+                return false;
+            }
+
+           // url and secret are valid
+           if($response->success()) {
```

Comment:
Why do we need this additional conditional?

Reply:
I restructured the code a bit. See latest commit